node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/identity/AthenzCredentialsMaintainer.java

```diff
@@ -0,0 +1,281 @@
+        SSLContext containerIdentitySslContext =
+                new SslContextBuilder()
+                        .withKeyStore(privateKeyFile.toFile(), certificateFile.toFile())
+                        .withTrustStore(trustStorePath.toFile(), KeyStoreType.JKS)
+                        .build();
+        try (ZtsClient ztsClient = new DefaultZtsClient(ztsEndpoint, containerIdentitySslContext)) {
+            InstanceIdentity instanceIdentity =
+                    ztsClient.refreshInstance(
+                            configserverIdentity,
+                            containerIdentity,
+                            instanceId.asDottedString(),
+                            false,
+                            csr);
+            writePrivateKeyAndCertificate(keyPair.getPrivate(), instanceIdentity.certificate());
+            log.info("Instance successfully refreshed and credentials written to file");
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        } catch (Exception e) {
+            // TODO Change close() in ZtsClient to not throw checked exception
+        }
+    }
+
+    private void writePrivateKeyAndCertificate(PrivateKey privateKey, X509Certificate certificate) throws IOException {
+        Path tempPrivateKeyFile = toTempPath(privateKeyFile);
+        Files.write(tempPrivateKeyFile, KeyUtils.toPem(privateKey).getBytes());
+        Path tempCertificateFile = toTempPath(certificateFile);
+        Files.write(tempCertificateFile, X509CertificateUtils.toPem(certificate).getBytes());
+
+        Files.move(tempPrivateKeyFile, privateKeyFile, StandardCopyOption.ATOMIC_MOVE);
+        Files.move(tempCertificateFile, certificateFile, StandardCopyOption.ATOMIC_MOVE);
+    }
+
+    private static Path toTempPath(Path file) {
+        return Paths.get(file.toAbsolutePath().toString() + ".tmp");
+    }
+
+    // TODO Move to vespa-athenz
+    private String toAttestationDataString(SignedIdentityDocument signedIdDoc) throws JsonProcessingException {
+        com.yahoo.vespa.athenz.identityprovider.api.IdentityDocument idDoc = signedIdDoc.identityDocument();
+        com.yahoo.vespa.athenz.identityprovider.api.bindings.IdentityDocument identityDocumentPayload =
+                new com.yahoo.vespa.athenz.identityprovider.api.bindings.IdentityDocument(
+                        ProviderUniqueId.fromVespaUniqueInstanceId(idDoc.providerUniqueId()),
+                        idDoc.configServerHostname(),
+                        idDoc.instanceHostname(),
+                        idDoc.createdAt(),
+                        idDoc.ipAddresses());
+        String rawIdentityDocument = objectMapper.writeValueAsString(identityDocumentPayload);
+        com.yahoo.vespa.athenz.identityprovider.api.bindings.SignedIdentityDocument payload =
+                new com.yahoo.vespa.athenz.identityprovider.api.bindings.SignedIdentityDocument(
```

Comment:
You were supposed to come up with some really good alternate names for the bindings classes?

Reply:
Yes, see other PR from Friday :) I cannot use the new classes here yet as it will require waiting for another Vespa release.